5.0/en/0x14-V6-Cryptography.md

```diff
@@ -2,62 +2,318 @@
+
+Any other cipher options MUST NOT be used.
+
+### Disallowed Ciphers
+
+The following is a list of _explicitly unauthorized_ ciphers. Whilst anything not on the above list must be approved with documented reasons, the following are explicitly banned and MUST NOT be used:
+
+| Symmetric Key Algorithms |
+|--|
+| 2TDEA   | 
+| 3TDEA (3DES)   |
+| IDEA    |
+| RC4     |
+| Blowfish|
+| ARC4    |
+| DES     |
+
+### AES Cipher Modes
+
+Modern ciphers make use of various modes, particularly AES  for various purposes. We describe the requirements on AES Cipher Modes here.
+
+#### Approved Cipher Modes for General Use Cases
+
+Implementations MUST choose from the following block modes, in order of preference, EXCEPT where the function is encrypted data storage (see next subsection):
+
+| AES Encryption Mode | Authenticated?* | Reference | L1 | L2 | L3 |
+|--|--|--|--|--|--|
+| GCM | Yes | [NIST SP 800-38D](https://csrc.nist.gov/pubs/sp/800/38/d/final) |  ✓  |  ✓  |  ✓  |
+| CCM | Yes | [NIST SP 800-38C](https://csrc.nist.gov/pubs/sp/800/38/c/upd1/final) |  ✓  |  ✓  |  ✓  |
+| CCM-8** | Yes | [RFC 6655](https://www.rfc-editor.org/info/rfc6655) |  ✓  |  ✓  |  ✓  |
+| CBC | No | [NIST SP 800-38A](https://csrc.nist.gov/pubs/sp/800/38/a/final) |  ✓  |  ✓  |  ✓  |
+
+* All encrypted messages must be authenticated. Given this, for ANY use of CBC mode there MUST be an associated hashing function or MAC to validate the message. This MUST be applied in the 'Encrypt-Then-Hash' or 'ETH' method. If this cannot be guaranteed, then CBC MUST NOT be used.
+
+**CCM-8 is included in regard to TLS cipher suites (see [TLS](https://github.com/santander-group-cyber-cto/CryptographyStandard/blob/main/Implementations/TLS/README.md) section). 
+
+#### Recommendations for Approved Cipher Modes for General Use Cases
+
+Out of the given approved block modes, implementations SHOULD use the ciphers in this list, in order of preference:
+
+| AES Encryption Mode | Reference | L1 | L2 | L3 |
+|--|--|--|--|--|
+| GCM | [NIST SP 800-38D](https://csrc.nist.gov/pubs/sp/800/38/d/final) |  ✓  |  ✓  |  ✓  |
+| CCM | [NIST SP 800-38C](https://csrc.nist.gov/pubs/sp/800/38/c/upd1/final) |  ✓  |  ✓  |  ✓  |
+
+#### Approved Cipher Modes ONLY for Data Storage (block encryption on disk)
+
+Implementations of disk-level block encryption MUST choose from the following list, in order of preference:
+
+| AES Disk Encryption Mode | Reference | L1 | L2 | L3 |
+|--|--|--|--|--|
+| XTS | [NIST SP 800-38E](https://csrc.nist.gov/pubs/sp/800/38/e/final) |  ✓  |  ✓  |  ✓  |
+| XEX | [Rogaway 2004](https://doi.org/10.1007/978-3-540-30539-2_2) |  ✓  |  ✓  |  ✓  |
+| LRW | [Liskov, Rivest, and Wagner 2005](https://doi.org/10.1007/s00145-010-9073-y) |  ✓  |  ✓  |  ✓  |
+
+#### Disallowed Cipher Modes
+
+The following cipher modes MUST NOT be used for any use case:
+
+| Encryption Mode | 
+|--|
+| ECB |
+| CFB | 
+| OFB | 
+| CTR | 
+
+### Key Wrapping
```

Comment:
Would there be need to explain for the layman when key wrapping should be used?

Reply:
100% agree with this, yes there is indeed a need and we are adding it